rosetta-source/src/main/rosetta/product-template-type.rosetta

```diff
@@ -413,31 +407,28 @@ type ConstituentWeight: <"A class describing the weight of each of the underlier
     schedule CalculationSchedule (0..1) <"Allows the full representation of a payout by defining a set of schedule periods. It supports standard schedule customization by expressing all the dates, quantities, and pricing data in a non-parametric way.">
+    
+    condition Underlier:  <"The underlier for a settlement payout cannot be a NonTransferableProduct.">
+        underlier -> Product -> NonTransferableProduct is absent
 
-    condition SettlementTerms: <"For foreign exchange contracts, the settlement terms must exist.">
-        if underlier -> index -> ForeignExchangeRate exists
-        then settlementTerms exists
+    condition Basket:  <"If the underlier is a basket, it must contain only transferable assets.">    
+        if underlier -> Observable -> Basket exists
+        then underlier -> Observable -> Basket -> basketConstituent -> Asset only exists
 
-    condition SettlementDate: <"For foreign exchange contracts, either the settlementDate is set or the cashflowDates, but not both. When the cashflowDates are set, they must be the same for the 2 legs of the currency pair.">
-        if underlier -> index -> ForeignExchangeRate exists
-        then (settlementTerms -> settlementDate -> valueDate exists
-                    and underlier -> foreignExchange -> exchangedCurrency1 -> settlementTerms -> settlementDate -> adjustableOrRelativeDate is absent
-                    and underlier -> foreignExchange -> exchangedCurrency2 -> settlementTerms -> settlementDate -> adjustableOrRelativeDate is absent)
-                or (settlementTerms -> settlementDate -> valueDate is absent
-                    and underlier -> foreignExchange -> exchangedCurrency1 -> settlementTerms -> settlementDate -> adjustableOrRelativeDate exists
-                    and underlier -> foreignExchange -> exchangedCurrency2 -> settlementTerms -> settlementDate -> adjustableOrRelativeDate exists
-                    and underlier -> foreignExchange -> exchangedCurrency1 -> settlementTerms -> settlementDate -> adjustableOrRelativeDate = underlier -> foreignExchange -> exchangedCurrency2 -> settlementTerms -> settlementDate -> adjustableOrRelativeDate)
+    condition Index:  <"If the underlier is an Index, then it can only be cash settled.">
+        if underlier -> Observable -> Index exists
+        then settlementTerms -> cashSettlementTerms exists 
 
-    condition FxSettlement: <"For foreign exchange contracts, the settlement type must be either fx non-deliverable settlement or not specified, which implies physical settlement in the case of foreign exchange.">
-        if underlier -> foreignExchange exists
-        then settlementTerms -> physicalSettlementTerms is absent
+    condition SettlementTerms: <"Settlement Terms should be defined except for cash.">
+        if underlier -> Observable -> Asset -> Cash is absent
+        then settlementTerms exists
```

Comment:
The current condition applies to foreign exchange contracts, so new logic should be:
```
if underlier -> Observable -> Asset -> Cash exists
```

Reply:
done